chat-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -95,11 +79,13 @@ export function makeAddMessageToConversationRoute({
     res: ExpressResponse<ApiMessage>
   ) => {
     const reqId = getRequestId(req);
+
     try {
       const {
         params: { conversationId: conversationIdString },
         body: { message },
         query: { stream },
+        headers: { origin: requestOrigin },
```

Comment:
could the logic around getting origin be middleware instead of baked into the route?

and then for persisting to the DB, i think we could make an injected function that takes the request object and adds arbitrary data from it (including origin) to the object in the DB.

Reply:
I could switch this to middleware yes. Not sure exactly what you're referring to with an injected function / why we need it. I figure we can just treat this like `ip` and pass it to the service?

Reply:
> I figure we can just treat this like ip and pass it to the service?

we shouldn't require that we pass the IP either tbh.

the server core library should be as agnostic as possible about all the non-essential information that we pass to it, leaving that up to each implementation. 

let's get this change into main as it is now, and then we'll refactor in the feature branch for OSSification.

Reply:
> and then for persisting to the DB, i think we could make an injected function that takes the request object and adds arbitrary data from it (including origin) to the object in the DB.

thinking a bit more about this, and rather than injecting function, i think we could just include `ip`, `origin`, and whatever other request data that we want to persist to a custom property on the Express `Request` object, say `Request.customData`. 

Then, in the `POST /conversation` route, we put whatever is in the `Request.customData` object in the Conversation document in the DB, say in a `Conversation.customData` object.